src/critics/local_goal_critic.cpp

```diff
@@ -0,0 +1,109 @@
+// Copyright 2022 @artofnothingness Alexey Budyakov, Samsung Research
+#include "mppic/critics/local_goal_critic.hpp"
+
+#include <xtensor/xmath.hpp>
+#include <xtensor/xsort.hpp>
+
+namespace mppi::critics
+{
+
+void LocalGoalCritic::initialize()
+{
+  auto getParam = parameters_handler_->getParamGetter(name_);
+
+  getParam(consider_angle_, "consider_angle_", true);
+  getParam(angle_offset_, "angle_offset", 6);
+  getParam(angle_cost_power_, "angle_cost_power", 1);
+  getParam(angle_cost_weight_, "angle_cost_weight", 3.0);
+
+  getParam(consider_distance_, "consider_distance", true);
+  getParam(distance_goal_count_, "disance_goal_count", 2);
```

Comment:
Both goal count and offset are both being applied as offsets, I think they should have some similarity in naming so its clear that they relate. E.g. distance_offset_lower and distance_offset_upper or something

Reply:
I'm not sure this appropriate here. offset is offset from furthest point, count is the number of points to consider. we not setting upper bound here. Although i did think about name unifications for such upper/lower bounds.... hmm

Reply:
I don't think `offset` implies non-number of points. Offset is such a generic word that if you just say in the readme that its the integer number of points, that's fine. I don't think it implies distance unless you say it does.

Reply:
renamed